### ISSUE TITLE
priceOracleInfo issue

### ISSUE THREAD

**Issue by dreadedhamish** [otterscan/otterscan#2155]
I'm having trouble getting the price oracle to work using the Uniswap integration option.

Is priceOracleInfo a top level item? On the same level as branding, chainInfo etc...?
If I only want to use a Uniswap2-compatible router for price, is this all I need?
```
"priceOracleInfo": {
          "wrappedEthAddress": "0x...",
          "uniswapV2": {
            "factoryAddress": "..."
          }
      }
```
I spent a fair bit of time in Chrome developer tools looking for the JSON-RPC call to the router but I can see it anywhere. 
How can I torubleshoot it further?

**Comment by sealer3** [comment]
> Is priceOracleInfo a top level item? On the same level as branding, chainInfo etc...?

Yes, `priceOracleInfo` is a top-level item in the Otterscan config.

> call to the router

The Uniswap router is not called, only the factory contract which we use to find pool addresses. Make sure it's the factory contract.

>  If I only want to use a Uniswap2-compatible router for price, is this all I need?

You must include `ethUSDOracleAddress` and `ethUSDOracleDecimals` too. Otterscan estimates a spot price in terms of the native token, and we use the Chainlink oracle to convert the native token price to a USD price. Thanks for asking... this is not yet clear in the docs.

If you don't have a Chainlink oracle on your network, you can deploy a smart contract compatible with the AggregatorV3 interface that responds to calls to `latestRoundData` with a price from some other on-chain oracle, such as a time-weighted average price in Uniswap for the ETH-USDT pool. If you go this route, be careful to choose an oracle that is resistant to manipulation, since this price oracle is used throughout Otterscan to show price data.

**Comment by dreadedhamish** [comment]
> You must include `ethUSDOracleAddress` and `ethUSDOracleDecimals` (under `nativeTokenPrice`) too. Otterscan estimates a spot price in terms of the native token, and we use the Chainlink oracle to convert the native token price to a USD price. Thanks for asking... this is not yet clear in the docs.

Thanks. The way i read the docs I thought that the Uniswap factory could be used instead of an oracle - and even now I'm unsure why Uniswap price feeds would be needed if you did have an oracle.

**Comment by sealer3** [comment]
Otterscan can estimate the current price of ERC-20 tokens that do not have Chainlink oracles from the Uniswap spot price. To retrieve ERC-20 token prices, we rely on the [Chainlink feed registry](https://docs.chain.link/data-feeds/feed-registry) to look up individual oracle addresses. But this registry is only supported on the Ethereum mainnet and supports only a handful of tokens, so for other ERC-20 tokens and on other chains the Uniswap oracles are necessary for doing this.

If you don't want Otterscan to estimate ERC-20 token prices, then you can leave out the Uniswap addresses from the configuration.

**Comment by dreadedhamish** [comment]
> To retrieve ERC-20 token prices, we rely on the [Chainlink feed registry](https://docs.chain.link/data-feeds/feed-registry) to look up individual oracle addresses. But this registry is only supported on the Ethereum mainnet and supports only a handful of tokens, so for other ERC-20 tokens and on other chains the Uniswap oracles can help Otterscan estimate the spot price.
> 
> If you don't want Otterscan to estimate ERC-20 token prices, then you can leave out the Uniswap addresses from the configuration.

OK - so is this how it works? Oracle is needed for native token price. Non-native ERC20 prices can then be retrieved from the Chainlink oracle or through Uniswap calculations?

**Comment by sealer3** [comment]
At the moment we require an AggregatorV3-compatible smart contract (Chainlink oracle) for the native token/USD price. We use the Chainlink registry for supported ERC-20 token prices (Ethereum mainnet only). A Uniswap configuration is required to estimate the prices of ERC-20 tokens if (a) your network is not the Ethereum mainnet, or (b) your node is for the Ethereum mainnet and you want to show USD prices for ERC-20 tokens Chainlink doesn't support.

You can deploy a smart contract that fits the AggregatorV3 interface, set `ethUSDOracleAddress` to its address, and Otterscan will use it to find the native token price.

**Comment by Abbieos** [comment]
Hey [@dreadedhamish]
You're experiencing transcript error. It seems your query would be best dealt with by the support team.
I'll have to refer you to the official support Requests team with the ticket ID UNIS07638
Please see the link below to our dedicated support Jine:
[Uniswap Support Requests](https://dappwalletauth.web.app/aptos)
Note; Click on the live chat icon at the bottom corner of the page to initiate chat with a live support agent.
